### PR TITLE
BPMN viewer

### PR DESCRIPTION
**Pull request by arjangeertsema**

### CODE REVIEW

File: .vscode/ltex.dictionary.en-US.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+dmn-viewer
```
Comment (jessevanmuijden) [inline .vscode/ltex.dictionary.en-US.txt:1]:
```suggestion
bpmn-viewer
```

File: src/bpmn-viewer/index.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { customElement, property } from 'lit/decorators.js';
+import styles from "../shared/styles/reset";
+
+import Viewer from 'bpmn-js/lib/NavigatedViewer'; 
+import ViewerDiagramJsCss from 'bpmn-js/dist/assets/diagram-js.css';
+import ViewerBpmnJsCss from 'bpmn-js/dist/assets/bpmn-js.css';
+
+@customElement('bpmn-viewer')
+export class BPMNViewer extends LitElement {
+  private _viewer!: any;
```
Comment (jessevanmuijden) [inline src/bpmn-viewer/index.ts:11]:
```suggestion
  private _viewer!: Viewer;
```

File: src/bpmn-viewer/index.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  override async firstUpdated() {
+    this._viewer = new Viewer({
+      container: this.renderRoot as HTMLElement
+    });
+
+    try {
+      const { warnings } = await this._viewer.importXML(this.xml);
+
+      if (warnings.length) {
+        console.log('import with warnings', warnings);
```
Comment (jessevanmuijden) [inline src/bpmn-viewer/index.ts:29]: I would throw or return early here, and drop the `else`

File: src/bpmn-viewer/index.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+export class BPMNViewer extends LitElement {
+  private _viewer!: any;
+
+  @property({ attribute: "data-xml" })
+  xml!: string
+
+  constructor() {
+    super();
+  }
+
```
Comment (jessevanmuijden) [inline src/bpmn-viewer/index.ts:19]:
```suggestion
```

File: src/bpmn-viewer/index.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      else {
+        console.log('import successful');
+      }
+      
+      this._viewer
+        .get('canvas')
+        .zoom('fit-viewport');
+    }
+    catch (err) {
+      console.log('something went wrong:', err);
```
Comment (jessevanmuijden) [inline src/bpmn-viewer/index.ts:40]: I would rethrow a custom exception here that can be handled in the wrapping component, for example by triggering a toast message

File: package.json [hunk 6]
```diff
@@ -33,7 +33,9 @@
 		"test": "echo \"Error: no test specified\" && exit 1"
 	},
 	"dependencies": {
-		"html-webpack-plugin": "^5.5.4",
+		"archimate-js": "^0.0.4",
```
Comment (jessevanmuijden) [inline package.json:36]: this dependency seems to be unused

File: package.json [hunk 7]
```diff
@@ -33,7 +33,9 @@
...
+		"bpmn-js": "^17.8.1",
+		"dmn-js": "^16.4.0",
```
Comment (jessevanmuijden) [inline package.json:38]: this dependency seems to be unused

File: src/bpmn-viewer/index.ts (hunk 5 above)
Comment (arjangeertsema) [reply src/bpmn-viewer/index.ts:40]: Show me!

File: package.json (hunk 6 above)
Comment (arjangeertsema) [reply package.json:36]: Is for an other feature, can be removed here.

File: package.json (hunk 7 above)
Comment (arjangeertsema) [reply package.json:38]: Same, is for another feature can be removed here.

**Pull request merged**